hw11/EvenApp.java

```diff
@@ -0,0 +1,47 @@
+package com.homework.hw11;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Scanner;
+
+public class EvenApp {
+    private static Scanner scanner;
+
+    public static void main(String[] args) {
+        scanner = new Scanner(System.in);
+        System.out.println(splitInputToArray());
+        System.out.println(dequeInput());
+
+        scanner.close();
+    }
+
+    private static String splitInputToArray() {
+        String[] arrayOfNumbers = scanner.nextLine().split(" ");
+        StringBuilder stringOfNumbers = new StringBuilder();
+        for (int i = arrayOfNumbers.length - 1; i >= 0; i--) {
+            if (i % 2 != 0) {
+                stringOfNumbers.append(arrayOfNumbers[i]).append(" ");
+            }
+        }
+        if (stringOfNumbers.length() > 0) {
+            stringOfNumbers.deleteCharAt(stringOfNumbers.length() - 1);
+        }
+        return String.valueOf(stringOfNumbers);
+    }
+
+    private static String dequeInput() {
+        Deque<Integer> dequeOfNumbers = new ArrayDeque<>();
+        int count = 0;
+        while (scanner.hasNextInt()) {
+            if (count % 2 != 0) {
+                dequeOfNumbers.add(scanner.nextInt());
```

Comment:
better write:
Integer number = scanner.nextInt();
```suggestion
                dequeOfNumbers.add(number);
```